src/python_testing/TCP_Tests.py

```diff
@@ -0,0 +1,150 @@
+
+        device.closeTCPConnectionWithPeer()
+        asserts.assert_equal(device.isActiveSession, False,
+                             "Large Payload Session should not be active after TCP connection closure")
+
+    @async_test_body
+    async def test_TCPConnectDisconnectThenConnectAgain(self):
+
+        try:
+            device = await self.default_controller.GetConnectedDevice(nodeid=self.dut_node_id, allowPASE=False, timeoutMs=1000,
+                                                                      payloadCapability=ChipDeviceCtrl.TransportPayloadCapability.LARGE_PAYLOAD)
+        except TimeoutError:
+            asserts.fail("Unable to establish a CASE session over TCP to the device")
+        asserts.assert_equal(device.isSessionOverTCPConnection, True, "Session does not have associated TCP connection")
+
+        device.closeTCPConnectionWithPeer()
+        asserts.assert_equal(device.isActiveSession, False,
+                             "Large Payload Session should not be active after TCP connection closure")
+
+        # Connect again
+        try:
+            device = await self.default_controller.GetConnectedDevice(nodeid=self.dut_node_id, allowPASE=False, timeoutMs=1000,
+                                                                      payloadCapability=ChipDeviceCtrl.TransportPayloadCapability.LARGE_PAYLOAD)
+        except TimeoutError:
+            asserts.fail("Unable to establish a CASE session over TCP to the device")
+        asserts.assert_equal(device.isSessionOverTCPConnection, True, "Session does not have associated TCP connection")
+        asserts.assert_equal(device.isActiveSession, True, "Large Payload Session should be active over TCP connection")
+
+    @async_test_body
+    async def test_OnOffToggleCommandOverTCPSession(self):
+
+        try:
+            device = await self.default_controller.GetConnectedDevice(nodeid=self.dut_node_id, allowPASE=False, timeoutMs=1000,
+                                                                      payloadCapability=ChipDeviceCtrl.TransportPayloadCapability.LARGE_PAYLOAD)
+        except TimeoutError:
+            asserts.fail("Unable to establish a CASE session over TCP to the device")
+        asserts.assert_equal(device.isSessionOverTCPConnection, True, "Session does not have associated TCP connection")
+        asserts.assert_equal(device.isActiveSession, True, "Large Payload Session should be active over TCP connection")
+        asserts.assert_equal(device.sessionAllowsLargePayload, True, "Session does not have associated TCP connection")
+
+        commands = Clusters.Objects.OnOff.Commands
+        try:
+            await self.send_single_cmd(cmd=commands.Toggle(), endpoint=1,
+                                       payloadCapability=ChipDeviceCtrl.TransportPayloadCapability.LARGE_PAYLOAD)
+        except InteractionModelError as e:
+            asserts.assert_equal(e.status, Status.Success, "Unexpected error returned")
```

Comment:
will an InteractionModelError always be thrown? If so should there be a fail if no exception is thrown?

The usage of a assert in an exception handler warrants a bit of a comment explaining why this is the case. I would expect sending a single commmand to not throw exceptions.

Reply:
I have replaced with an asserts.fail for the assert check in the exception handling block.